src/libValidator/Validator.cpp

```diff
@@ -17,17 +17,70 @@
 #include <vector>
 
 #include "Validator.h"
-#include "libData/AccountData/Account.h"
 #include "libMediator/Mediator.h"
 
 using namespace std;
 using namespace boost::multiprecision;
 
-Validator::Validator(Mediator& mediator)
+MediatorAdapter::MediatorAdapter(Mediator& mediator)
     : m_mediator(mediator)
 {
 }
 
+MediatorAdapter::~MediatorAdapter() {}
+
+unsigned int MediatorAdapter::getShardID() const
+{
+    return m_mediator.m_node->getShardID();
+}
+
+unsigned int MediatorAdapter::getNumShards() const
+{
+    return m_mediator.m_node->getNumShards();
+}
+const char* MediatorAdapter::currentEpochNumAsString() const
+{
+    return to_string(m_mediator.m_currentEpochNum).c_str();
+}
+
+NoopMediator::~NoopMediator() {}
```

Comment:
Move all `NoopMediator` code to test (if it's not used in actual code)

Reply:
Great idea